relay-general/src/protocol/breakdowns.rs

```diff
@@ -0,0 +1,339 @@
+                None => OperationBreakdown::DoNotEmit(operation_name),
+                Some(operation_name) => OperationBreakdown::Emit(operation_name),
+            };
+
+            intervals
+                .entry(operation_name)
+                .or_insert_with(Vec::new)
+                .push(cover);
+        }
+
+        if intervals.is_empty() {
+            return None;
+        }
+
+        let mut breakdown = Measurements::default();
+
+        let mut total_time_spent: u64 = 0;
+
+        for (operation_name, intervals) in intervals {
+            let op_time_spent: u64 = match get_op_time_spent(intervals) {
+                None => continue,
+                Some(op_time_spent) => op_time_spent,
+            };
+
+            total_time_spent += op_time_spent;
+
+            let operation_name = match operation_name {
+                OperationBreakdown::DoNotEmit(_) => continue,
+                OperationBreakdown::Emit(operation_name) => operation_name,
+            };
+
+            let time_spent_measurement = Measurement {
+                value: Annotated::new(op_time_spent),
+            };
+
+            let op_breakdown_name = format!("ops.{}", operation_name);
+
+            breakdown.insert(op_breakdown_name, Annotated::new(time_spent_measurement));
+        }
+
+        let total_time_spent_measurement = Measurement {
+            value: Annotated::new(total_time_spent),
+        };
+        breakdown.insert(
+            "total.time".to_string(),
+            Annotated::new(total_time_spent_measurement),
+        );
+
+        Some(breakdown)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Empty, ProcessValue)]
+pub enum BreakdownMeasurements {
+    U64(Measurements<u64>),
+}
+
+impl Deref for BreakdownMeasurements {
+    type Target = Measurements<u64>;
+
+    fn deref(&self) -> &Self::Target {
+        match self {
+            BreakdownMeasurements::U64(x) => x,
+        }
+    }
+}
+
+impl FromValue for BreakdownMeasurements {
+    fn from_value(value: Annotated<Value>) -> Annotated<Self> {
+        match Measurements::<u64>::from_value(value) {
+            Annotated(None, meta) => Annotated(None, meta),
+            Annotated(Some(value), meta) => {
+                Annotated(Some(BreakdownMeasurements::U64(value)), meta)
+            }
+        }
+    }
+}
+
+impl ToValue for BreakdownMeasurements {
+    fn to_value(self) -> Value {
+        match self {
+            BreakdownMeasurements::U64(x) => x.to_value(),
+        }
+    }
+
+    fn serialize_payload<S>(&self, s: S, behavior: SkipSerialization) -> Result<S::Ok, S::Error>
+    where
+        Self: Sized,
+        S: Serializer,
+    {
+        match self {
+            BreakdownMeasurements::U64(x) => x.serialize_payload(s, behavior),
+        }
+    }
+}
+
+/// Configuration to define breakdown to be generated based on properties and breakdown type.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(tag = "type", rename_all = "camelCase")]
```

Comment:
Let me double-check, but I *think* we use `snake_case` for discriminators.

Reply:
I'll use `snake_case`. I took `camelCase` based on its usage in the PII configurations.